File: MyDent.Services/Database/News.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MyDent.Services.Database
{
    public class News
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(200)]
        public string Title { get; set; } = string.Empty;

        [Required]
        public string Content { get; set; } = string.Empty;

        public int? ImageAssetId { get; set; }

        [ForeignKey("ImageAssetId")]
        public Asset? Image { get; set; }

        public bool IsPublished { get; set; } = true;

        public DateTime PublishedAt { get; set; } = DateTime.UtcNow;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public int CreatedByUserId { get; set; }

        [ForeignKey("CreatedByUserId")]
        public User CreatedByUser { get; set; } = null!;
    }
}

File: UI/mydent_desktop/lib/models/news.dart
import 'package:json_annotation/json_annotation.dart';

part 'news.g.dart';

@JsonSerializable()
class News {
  final int? id;
  final String? title;
  final String? content;
  final int? imageAssetId;
  final bool? isPublished;
  final DateTime? publishedAt;
  final DateTime? createdAt;
  final int? createdByUserId;
  final String? createdByUserName;

  News({
    this.id,
    this.title,
    this.content,
    this.imageAssetId,
    this.isPublished,
    this.publishedAt,
    this.createdAt,
    this.createdByUserId,
    this.createdByUserName,
  });

  factory News.fromJson(Map<String, dynamic> json) => _$NewsFromJson(json);

  Map<String, dynamic> toJson() => _$NewsToJson(this);
}
